src/Adapter/Form/ChoiceProvider/InstalledPaymentModulesChoiceProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Form\ChoiceProvider;
+
+use Module;
+use PaymentModule;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use Validate;
+
+final class InstalledPaymentModulesChoiceProvider implements FormChoiceProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getChoices(): array
+    {
+        $paymentModules = [];
```

Comment:
Store $paymentModules in property, to be sure the foreach will be not executed twice if we have two `InstalledPaymentModulesChoiceProvider` in the same page.

Reply:
Hmm, wouldn't that make service stateful? 

Or image we have this kind of flow:
1. Call InstalledPaymentModulesChoiceProvider
2. Install payment method
3. Call InstalledPaymentModulesChoiceProvider

Reply:
Another possible scenario.
You have a `CollectionType` with a `InstalledPaymentModulesChoiceProvider`, you have 10 rows. This means, you call the same code 10 times, in the same form :thinking: 

If you have a form and you install a payment method in GET method, it's a bad idea.
If you have a form and generate the form before saving payment method, and after regenerate the code, it's a bad idea too.

Normal scenario:
1. GET: Generate form with `InstalledPaymentModulesChoiceProvider`
2. POST. install payment method
3. REDIRECT
4. GET: Generate form with `InstalledPaymentModulesChoiceProvider`

Reply:
There can be many scenarios, but I'll store it in static attribute. :)